core-bundle/tests/String/HtmlAttributesTest.php

```diff
@@ -366,6 +424,70 @@ public function testDoesNotOutputEmptyClassAttribute(): void
         $this->assertSame('', $attributes->toString());
     }
 
+    public function testAddAndRemoveStyles(): void
+    {
+        // Whitespaces should get normalized by default
+        $attributes = new HtmlAttributes(['style' => " \ffoo:  bar;\tother1\n:other2\r  ;"]);
+
+        $this->assertSame('foo:bar;other1:other2', $attributes['style']);
+
+        // And remove classes
+        $attributes->addStyle('baz: 1');
+        $attributes->addStyle('foo: red; foobar: baz');
+        $attributes->addStyle(['foo2: bar', 'foo3: bar; foo4: bar', 'foobar' => 'red']);
+        $attributes->removeStyle('other2');
+        $attributes->removeStyle('thing: x; other1: red;');
+        $attributes->removeStyle(['foo3 : x', ' foo4 ']);
+
+        $this->assertSame('foo:red;baz:1;foobar:red;foo2:bar', $attributes['style']);
+    }
+
+    public function testAddAndRemoveConditionalStyles(): void
+    {
+        $attributes = new HtmlAttributes();
+
+        $attributes->addStyle('foo: a', null);
+        $attributes->addStyle('foo: b', false);
+        $attributes->addStyle('foo: c', 0);
+        $attributes->addStyle('foo: d', '');
+
+        $this->assertSame([], iterator_to_array($attributes));
+
+        $attributes->addStyle('a: foo', condition: true);
+        $attributes->addStyle('b: foo', condition: 1);
+        $attributes->addStyle('c: foo', condition: 'true');
+        $attributes->addStyle('d: foo', condition: '1');
+
+        $this->assertSame(['style' => 'a:foo;b:foo;c:foo;d:foo'], iterator_to_array($attributes));
+
+        $attributes->removeStyle('a: foo', null);
+        $attributes->removeStyle('b: foo', false);
+        $attributes->removeStyle('c', 0);
+        $attributes->removeStyle('d: foo', '');
+
+        $this->assertSame(['style' => 'a:foo;b:foo;c:foo;d:foo'], iterator_to_array($attributes));
+
+        $attributes->removeStyle('a: foo', condition: true);
+        $attributes->removeStyle('b', condition: 1);
+        $attributes->removeStyle('c: foo', condition: 'true');
+        $attributes->removeStyle(['d'], condition: '1');
+
+        $this->assertSame([], iterator_to_array($attributes));
+    }
+
+    public function testDoesNotOutputEmptyStyleAttribute(): void
+    {
+        $attributes = new HtmlAttributes();
+        $attributes->addStyle('');
+
+        $this->assertSame('', $attributes->toString());
+
+        $attributes->addStyle('foo');
```

Comment:
Shouldn‘t styles without value be forbidden? They are never valid, are they?

Reply:
IMHO, the class should be as forgiving as possible in the fluent API. So that you do not have to handle special cases like this in your templates:

```twig
{# c can potentially be empty - now you would need to break the fluent calls #}
{{ attrs().addClass(c).addStyle(s).set('foo', 'bar') }}
```

An empty style/class attribute should not be output, though.

Reply:
That‘s not an answer to my question. I agree that `s` can be empty but it cannot be just `color` without a value.

Reply:
Ah, that's what you meant. Should we maybe infer the value `reset` then?

Reply:
This test code was wrong (because this line is acutally a no-op), fixed in 863d4d71cb27b65c9609d10505996988a94d2a1f

> Shouldn‘t styles without value be forbidden?

You mean by throwing an exception? I don’t think we should.
Currently invalid styles are just ignored.

Reply:
> Should we maybe infer the value `reset` (or `initial`) then?

I don’t think we should. Why would you want to omit the `:` in any case? Empty values like `color:;` are already valid CSS.